### ISSUE TITLE
Peer_manager requires sanity check for added peer address

### ISSUE THREAD

**Issue by SWvheerden** [tari-project/tari#5373]
The peer manager needs to check the address it gets from other peers. 
If we just allow any invalid address, tor windows gets flooded with this:
```
 [warn] Invalid onion hostname [scrubbed]; rejecting
[warn] Service address [scrubbed] invalid checksum.
```

**Comment by stringhandler** [comment]
Do you have an example of a bad address? It'll help to test the use case

**Comment by SWvheerden** [comment]
@sdbondi Can you supply a few of the examples please

**Comment by sdbondi** [comment]
Specifically, the onion address checksum should be checked. https://github.com/torproject/torspec/blob/main/rend-spec-v3.txt#LL2258C2-L2258C2

This should probably be done in the `multiaddr` crate

**Comment by brianp** [comment]
I'm going to say, I disagree the added code is worth maintaining considering how little it protects from. We could validate a tor address is built correctly but tor addresses aren't the only valid addresses. We take local TCP connections etc as well. And even if it's a valid address format doesn't make it good.

I think the sync limiter actually helps a lot here. Because if a bad actor populates a bunch of crap. I sync all the crap, I still won't send it on if I can't connect to any of it. So it can never propagate. The bad actor can only ever send it to direct connections, and all of those people will never send it on because they'll never come online. The limiter reduces the surface area better than a format validator could.

**Comment by SWvheerden** [comment]
this is now checked

**Comment by sdbondi** [comment]
Fixed in #5440
